overcast/runner/__init__.py

```diff
@@ -400,6 +443,24 @@ def provision_step(self, details):
         else:
             userdata = None
 
+        pending_nodes = set()
+        done          = set()
+
+        if details.get('retry-delay', False):
+            retry_delay = utils.parse_time(details['retry-delay'])
+        else:
+            retry_delay = 5
+
+        if details.get('timeout', False):
+            provision_timeout = utils.parse_time(details['timeout'])
+            deadline =  time.time() + provision_timeout
+        else:
+            provision_timeout = None
+            deadline = None
```

Comment:
I have thought of same logic as shell_step here, provisioning step can have retry_if_fail and retry_delay, also can have a limit to how long you can retry - instead of setting a number of retry setting, I just say "you can retry with retry_delay as delay btn retries upto  timeout value of provisioning step as I see similar logic in shell_step.  

Setting num_retry rather than only relying on timeout can be a better idea, so we can have num_retry globally and if you want, it can be set in individual steps too.